src/Libraries/Analysis/SurfaceData.cs

```diff
@@ -186,24 +188,31 @@ public void Tessellate(IRenderPackage package, double tol = -1, int maxGridLines
             var colorCount = 0;
             var uvCount = 0;
 
-            for (var i = 0; i < package.TriangleVertices.Count; i += 3)
+            var uvs = package.MeshTextureCoordinateBuffer.ToList();
+            var colors = package.MeshVertexColorBuffer.ToList();
+
+            var newColors = new byte[colors.Count()];
+
+            for (var i = 0; i < colors.Count(); i += 4)
             {
-                var uvu = package.TriangleUVs[uvCount];
-                var uvv = package.TriangleUVs[uvCount + 1];
+                var uvu = uvs.ElementAt(uvCount);
+                var uvv = uvs.ElementAt(uvCount + 1);
 
```

Comment:
This is using `ToList()` and then using `ElementAt` and `Count()`.  There's really no reason you have to build these arrays other than the `IRenderPackage.ApplyMeshVertexColors(byte[])` method.